website/project/views/node.py

```diff
@@ -1042,9 +1044,15 @@ def project_generate_private_link_post(auth, node, **kwargs):
 
     has_public_node = any(node.is_public for node in nodes)
 
-    new_link = new_private_link(
-        name=name, user=auth.user, nodes=nodes, anonymous=anonymous
-    )
+    try:
+        new_link = new_private_link(
```

Comment:
Can you add unit tests for naming and editing the private link names if there aren't any already?